joblib/test/test_parallel.py

```diff
@@ -2008,3 +2009,56 @@ def parallel_call(n_jobs):
         parallel_call(n_jobs)
         executor = get_reusable_executor(reuse=True)
         assert executor == first_executor
+
+
+@with_multiprocessing
+@parametrize('n_jobs', [2, 4, -1])
+@parametrize('backend', PROCESS_BACKENDS)
+@parametrize("context", [parallel_config, parallel_backend])
+def test_initializer(n_jobs, backend, context):
+    n_jobs = effective_n_jobs(n_jobs)
+    manager = mp.Manager()
+    queue = manager.list()
+
+    def initializer(queue):
+        queue.append("spam")
+
+    with context(
+        backend=backend,
+        n_jobs=n_jobs,
+        initializer=initializer,
+        initargs=(queue,)
+    ):
+        with Parallel() as parallel:
+            values = parallel(delayed(square)(i) for i in range(n_jobs))
+
+    assert len(queue) == n_jobs
+    assert all(q == "spam" for q in queue)
```

Comment:
@ogrisel I'm actually not sure what assertion I can make here (the current assertion seems not always to be true).

I want to test that every process that was started has run its init function.

However, it seems I sometimes reach line #2035 before all processes have been initialized - and by that time, the work has completed by fewer than the number of started processes.

Do you have any suggestions? Thanks!

Reply:
Hi @ogrisel -- just a gentle ping about this if you have any suggestions; thanks!